swarm/pss/forwarding_test.go

```diff
@@ -0,0 +1,183 @@
+package pss
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/ethereum/go-ethereum/p2p/protocols"
+	"github.com/ethereum/go-ethereum/swarm/network"
+	"github.com/ethereum/go-ethereum/swarm/pot"
+	whisper "github.com/ethereum/go-ethereum/whisper/whisperv5"
+)
+
+func TestForwardBasic(t *testing.T) {
+	base := newBaseAddress() // 0xFFFFFF.......
+	var peerAddresses []pot.Address
+	var dst pot.Address
+	const depth = 9
+	for i := 0; i <= depth; i++ {
```

Comment:
Maybe a comment on why we are adding two random addresses per depth?

Reply:
> Add two peers for each proximity order (same as in live system)

Still not sure why we are adding two peers....

In terms of _depth_ only one is required in each bin. In terms of _saturation_ two are needed. Could you please explain why we need to fulfill criterium of _saturation_ and not _depth_ here?